src/main/java/com/google/devtools/build/lib/bazel/rules/cpp/BazelCppSemantics.java

```diff
@@ -42,12 +42,16 @@ public PathFragment getEffectiveSourcePath(Artifact source) {
   @Override
   public void finalizeCompileActionBuilder(
       RuleContext ruleContext, CppCompileActionBuilder actionBuilder) {
-    actionBuilder.setCppConfiguration(ruleContext.getFragment(CppConfiguration.class));
+    CppConfiguration cppConfiguration = ruleContext.getFragment(CppConfiguration.class);
+    actionBuilder.setCppConfiguration(cppConfiguration);
     actionBuilder.setActionContext(CppCompileActionContext.class);
-    // Because Bazel does not support include scanning, we need the entire crosstool filegroup,
-    // including header files, as opposed to just the "compile" filegroup.
-    actionBuilder.addTransitiveMandatoryInputs(CppHelper.getToolchain(ruleContext).getCrosstool());
-    actionBuilder.setShouldScanIncludes(false);
+    boolean shouldScanIncludes = cppConfiguration.shouldScanIncludes();
+    if (!shouldScanIncludes) {
+      // When not performing include scanning, we need the entire crosstool filegroup,
+      // including header files, as opposed to just the "compile" filegroup.
+      actionBuilder.addTransitiveMandatoryInputs(CppHelper.getToolchain(ruleContext).getCrosstool());
```

Comment:
I will remove the if above.  While we have a number of compiler variation selections, we've done all of the legwork for cross/multi-target products in skylark proper, owing to the poor documentation and constrained observable behavior in CROSSTOOL </shameless-req-for-additional-docs>.